packages/react-bindings/src/styles/resolveStyles.ts

```diff
@@ -52,20 +52,53 @@ const resolveStyles = (
     disableAnimations,
     renderer,
     performance,
-  } = options || {}
+  } = options
 
   const { className, design, styles, variables, ...stylesProps } = props
-  const noInlineOverrides = !(design || styles || variables)
 
-  const cacheEnabled = performance.enableStylesCaching && noInlineOverrides
+  const noInlineStylesOverrides = !(design || styles)
+  const noVariableOverrides = performance.enableHardVariablesCaching || !variables
+
+  /* istanbul ignore else */
+  if (process.env.NODE_ENV !== 'production') {
+    if (!performance.enableStylesCaching && performance.enableHardVariablesCaching) {
+      throw new Error(
+        '@fluentui/react: Please check your "performance" settings on "Provider", to enable "enableHardVariablesCaching" you need to enable "enableStylesCaching"',
+      )
+    }
+
+    if (performance.enableHardVariablesCaching && variables) {
+      if (!_.isPlainObject(variables)) {
+        throw new Error(
+          '@fluentui/react: With "enableHardVariablesCaching" only plain objects can be passed to "variables" prop.',
+        )
+      }
+
+      const hasOnlyBooleanVariables = Object.keys(variables).every(variableName => {
+        return (
+          variables[variableName] === null ||
+          typeof variables[variableName] === 'boolean' ||
+          typeof variables[variableName] === 'undefined'
+        )
+      })
+
+      if (!hasOnlyBooleanVariables) {
```

Comment:
Should we just not cache for this instances of components which donot have only `boolean` variables instead of throwing an error? We are not throwing if caching is enabled and there are inline overrides, I don't see how this is different.

Reply:
The main goal to throw:
- easy detect broken cases
- avoid having these conditions in production as here we are iterating via object keys for example